v3/src/lib/dnd-kit/dnd-detect-collision.ts

```diff
@@ -38,18 +39,26 @@ export const dndDetectCollision: CollisionDetection = (_args) => {
   }
 
   // check for registered tile-specific collision handlers
-  for (const collision of collisions) {
-    const { id: collisionId } = collision
-    // find the first tile handler overlay that matches the collision
-    const handler = gTileCollisionDetectionRegistry.find(({overlayRegex}) => overlayRegex.test(`${collisionId}`))
-    if (handler) {
-      const { droppableRegex, detect } = handler
-      // filter the drop zones to those appropriate for the relevant tile
-      const containers = sortedContainers.filter(({id: containerId}) => droppableRegex.test(`${containerId}`))
-      // apply the collection detection function specified by the tile
-      return detect({ ...args, droppableContainers: containers })
+  if (sortedCollisions.length > 0) {
+    // find the prefix for the first tile collision
+    const prefix = `${sortedCollisions[0].id}`.match(prefixRegex)?.[0]
+    if (prefix) {
+      // find all collisions for the first tile
+      const tileCollisions = sortedCollisions.filter(c => `${c.id}`.startsWith(prefix))
+      for (const collision of tileCollisions) {
+        const { id: collisionId } = collision
+        const handler = gTileCollisionDetectionRegistry.find(({overlayRegex}) => overlayRegex.test(`${collisionId}`))
+        if (handler) {
+          const { droppableRegex, detect } = handler
+          // filter the drop zones to those appropriate for the relevant tile
+          const containers =
+            args.droppableContainers.filter(({id: containerId}) => droppableRegex.test(`${containerId}`))
+          // apply the collection detection function specified by the tile
```

Comment:
```suggestion
          // apply the collision detection function specified by the tile
```
Pre-existing condition, but may as well fix it since you're in the neighborhood. 😉